core/src/test/java/org/apache/gravitino/storage/TestSQLScripts.java

```diff
@@ -18,164 +18,139 @@
 
       } else if (upgradeMatcher.matches()) {
         String fromVersion = upgradeMatcher.group(1);
+        Assertions.assertTrue(
+            versionScrips.containsKey(fromVersion), "No schema script found for " + fromVersion);
 
-        Connection conn = versionConnections.get(fromVersion);
-        Assertions.assertNotNull(
-            conn, "No existing database connection found for version " + fromVersion);
-        Assertions.assertDoesNotThrow(
-            () -> executeSQLScript(conn, scriptFile),
-            "Failed to execute upgrade script" + " in file " + scriptFile.getName());
       } else if (metricsMatcher.matches()) {
-        // ignore iceberg metrics scripts for now
+        String version = metricsMatcher.group(1);
+        versionScrips.computeIfAbsent(version, k -> new ArrayList<>()).add(scriptFile);
+
       } else {
         Assertions.fail("Unrecognized script file name: " + scriptFile.getName());
       }
     }
+
+    for (List<File> scripts : versionScrips.values()) {
+      dropAllTables();
+      for (File scriptFile : scripts) {
+        String sqlContent = FileUtils.readFileToString(scriptFile, "UTF-8");
+        List<String> ddls =
+            Arrays.stream(sqlContent.split(";"))
+                .map(String::trim)
+                .filter(s -> !s.isEmpty())
+                .toList();
+
+        try (SqlSession sqlSession =
+            SqlSessionFactoryHelper.getInstance().getSqlSessionFactory().openSession(true)) {
+          try (Connection connection = sqlSession.getConnection()) {
+            try (Statement statement = connection.createStatement()) {
+              for (String ddl : ddls) {
+                Assertions.assertDoesNotThrow(
+                    () -> statement.execute(ddl),
+                    "Failed to execute DDL in file " + scriptFile.getName() + "ddl: " + ddl);
+              }
+            }
+          }
+        }
+      }
+    }
   }
 
-  private void executeSQLScript(Connection connection, File scriptFile) throws IOException {
-    String sqlContent = FileUtils.readFileToString(scriptFile, "UTF-8");
-    Arrays.stream(sqlContent.split(";"))
-        .map(String::trim)
-        .filter(s -> !s.isEmpty())
-        .forEach(
-            sql -> {
-              try (Statement stmt = connection.createStatement()) {
-                stmt.execute(sql);
-              } catch (SQLException e) {
-                throw new RuntimeException("Failed to execute SQL: " + sql, e);
+  private void dropAllTables() throws SQLException {
+    try (SqlSession sqlSession =
+        SqlSessionFactoryHelper.getInstance().getSqlSessionFactory().openSession(true)) {
+      try (Connection connection = sqlSession.getConnection()) {
+        try (Statement statement = connection.createStatement()) {
+          if ("postgresql".equals(backendType)) {
+            dropAllTablesForPostgreSQL(connection);
+          } else {
+            String query = "SHOW TABLES";
```

Comment:
Better to extract a method named `dropAllTablesInH2OrMySQL` or something similar.